File: playscript-cpp/src/grammar/BlockScope.cxx
#include "BlockScope.h"

using namespace play;

BlockScope::BlockScope() {
  this->name = "block" + index++;
}

BlockScope::BlockScope(Scope* enclosingScope, antlr4::ParserRuleContext* ctx) {
  this->name = "block" + index++;
  this->enclosingScope = enclosingScope;
  this->ctx = ctx;
}

std::string BlockScope::toString() {
  return "block " + name;
}

File: playscript-js/src/BlockScope.js
const Scope = require('./Scope')

class BlockScope extends Scope {
  static index = 1

  constructor(enclosingScope, ctx) {
    super()
    this.name = 'block' + BlockScope.index++
    if (enclosingScope && ctx) {
      this.enclosingScope = enclosingScope
      this.ctx = ctx
    }
  }

  toString() {
    return 'Block ' + this.name
  }
}

module.exports = BlockScope
